tools/donate_cpu_lib.py

```diff
@@ -0,0 +1,478 @@
+            sock.close()
+            print('Results have been successfully uploaded.')
+            return True
+        except socket.error as err:
+            print('Upload error: ' + str(err))
+            if retry < (max_retries - 1):
+                print('Retrying upload in 30 seconds')
+                time.sleep(30)
+    print('Upload permanently failed!')
+    return False
+
+
+def upload_info(package, info_output, server_address):
+    print('Uploading information output.. ' + str(len(info_output)) + ' bytes')
+    max_retries = 3
+    for retry in range(max_retries):
+        try:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock.connect(server_address)
+            send_all(sock, 'write_info\n' + package + '\n' + info_output + '\nDONE')
+            sock.close()
+            print('Information output has been successfully uploaded.')
+            return True
+        except socket.error as err:
+            print('Upload error: ' + str(err))
+            if retry < (max_retries - 1):
+                print('Retrying upload in 30 seconds')
+                time.sleep(30)
+    print('Upload permanently failed!')
+    return False
+
+
+def get_libraries():
+    libraries = ['posix', 'gnu']
+    library_includes = {'boost': ['<boost/'],
+                       # 'cairo': ['<cairo.h>'], <= enable after release of version 1.90
+                       'cppunit': ['<cppunit/'],
+                       'googletest': ['<gtest/gtest.h>'],
+                       'gtk': ['<gtk/gtk.h>', '<glib.h>', '<glib/', '<gnome.h>'],
+                       # 'kde': ['<KGlobal>', '<KApplication>', '<KDE/'], <= enable after release of version 1.90
+                       'libcerror': ['<libcerror.h>'],
+                       'libcurl': ['<curl/curl.h>'],
+                       'lua': ['<lua.h>', '"lua.h"'],
+                       'microsoft_sal': ['<sal.h>'],
+                       'motif': ['<X11/', '<Xm/'],
+                       'nspr': ['<prtypes.h>', '"prtypes.h"'],
+                       'opengl': ['<GL/gl.h>', '<GL/glu.h>', '<GL/glut.h>'],
+                       'openmp': ['<omp.h>'],
+                       'python': ['<Python.h>', '"Python.h"'],
+                       'qt': ['<QApplication>', '<QList>', '<qlist.h>', '<QObject>', '<QString>', '<qstring.h>', '<QWidget>', '<QtWidgets>', '<QtGui'],
+                       'ruby': ['<ruby.h>', '<ruby/', '"ruby.h"'],
+                       'sdl': ['<SDL.h>', '<SDL/SDL.h>', '<SDL2/SDL.h>'],
+                       'sqlite3': ['<sqlite3.h>', '"sqlite3.h"'],
+                       'tinyxml2': ['<tinyxml2', '"tinyxml2'],
+                       'wxwidgets': ['<wx/', '"wx/'],
+                       'zlib': ['<zlib.h>'],
+                      }
+    for library, includes in library_includes.items():
+        if has_include('temp', includes):
+            libraries.append(library)
+    return libraries
+
+
+my_script_name = os.path.splitext(sys.argv[0])[0]
```

Comment:
For me, it works under Debian Linux but not in Cygwin on Windows.
The reason is that `sys.argv[0]` sometimes contains only the name of the script and sometimes the full path of the script.
This depends on the OS (and maybe other things), see https://docs.python.org/3.7/library/sys.html#sys.argv
A path like `/home/user/cppcheck-/cygdrive/e/Files/GitHub/cppcheck/tools/donate-cpu-workfolder` is problematic here.

Reply:
Ok, I'll take a look of this problem

Reply:
https://github.com/danmar/cppcheck/pull/2294